lib/pure/ospaths.nim

```diff
@@ -662,6 +662,253 @@ when defined(windows) or defined(posix) or defined(nintendoswitch):
+  return false
+
+proc getRelativePathFromRelative(path, baseDir, curDir: string): string {.
+  noSideEffect.} =
+  ## Convert 'path' to a path relative to baseDir.
+  ##
+  ## Both 'path' and 'baseDir' must be relative paths from 'curDir'.
+  ## This proc never read filesystem.
+  ## 'baseDir' is always assumed to be a directory even if that path is actually a file.
+
+  proc skipCurDir(path: string): int {.noSideEffect.} =
+    var p = 0
+    let l = path.len
+    while p < l:
+      if p <= l - ParDir.len and continuesWith(path, ParDir, p):
+        break
+      if path[p] != CurDir:
+        break
+      inc(p)
+      p = skipDirSep(path, p, l)
+    return p
+
+  assert(not (isAbsolute(path) or isAbsolute(baseDir)))
+
+  if baseDir.len == 0:
+    return path
+
+  let alast = path.len
+  let blast = rSkipDirSep(baseDir, baseDir.len - 1, 0) + 1
+
+  let
+    astart = skipCurDir(path)
+    bstart = skipCurDir(baseDir)
+
+  var
+    apos = astart
+    bpos = bstart
+
+  while apos < alast and bpos < blast:
+    if not cmpCharInPath(path[apos], baseDir[bpos]):
+      break;
+    inc(apos)
+    inc(bpos)
+
+  if (bpos == blast and (apos == alast or path[apos].isSep)) or
+     (apos == alast and (bpos == blast or baseDir[bpos].isSep)):
+    inc(apos)
+  else:
+    while apos != astart and not path[apos-1].isSep:
+      dec(apos)
+      dec(bpos)
+
+  var numPar: int
+  (numPar, bpos) = countParDir(baseDir, bpos, blast)
+
+  let numUp = countDir(baseDir, bpos, blast)
+
+  if numPar == 0 and numUp == 0 and apos >= alast:
+    return $CurDir
+
+  result = if numUp > 0: ParDir & (DirSep & ParDir).repeat(numUp-1) else: ""
+
+  if numPar > 0:
+    if curDir.len == 0:
+      raise newException(ValueError, "parameter `curDir` is required to calculate relative path from given paths")
+    var cpos = curDir.len-1
+    for i in countDown(numPar-1, 0):
+      cpos = rSkipDirSep(curDir, cpos)
+      if isInRootDir(curDir, cpos) or curDir[cpos] == CurDir:
+        raise newException(ValueError, "Cannot calculate relative path from given paths")
+      while cpos > 0 and not curDir[cpos].isSep:
+        dec(cpos)
+    if curDir[cpos].isSep:
+      inc(cpos)
+    result = result / curDir.substr(cpos)
+
+  if apos < path.len:
+    return result / path.substr(apos)
+
+proc relativePath*(path, baseDir: string; curDir: string = ""): string {.
+  noSideEffect, rtl, extern: "nos$1".} =
+  ## Convert `path` to a path relative to baseDir.
+  ##
+  ## `path` and `baseDir` must be absolute paths or relative paths from `curDir`.
+  ## When one of `path` and `baseDir` is relative and other one is absolute, `curDir` must be absolute.
+  ##
+  ## On DOS like filesystem, when a drive of `path` is different from `baseDir`,
+  ## this proc just return the `path` as is because no way to calculate the relative path.
+  ##
+  ## This proc never read filesystem.
+  ## `baseDir` is always assumed to be a directory even if that path is actually a file.
+  runnableExamples:
```

Comment:
as @dom96 would say, there are too many tests here; it's now mentioned in style guide: https://nim-lang.github.io/Nim/contributing.html
exhaustive tests go to tospaths.nim
runnableExamples should show the minimum set that's representative of how the API works

Reply:
@demotomohiro you can just write "done" and close as resolved :-)

Reply:
done